pkg/plugin/GlobalPluginService.go

```diff
@@ -1653,3 +1660,266 @@ func (impl *GlobalPluginServiceImpl) deletePlugin(pluginDeleteReq *PluginMetadat
+func (impl *GlobalPluginServiceImpl) MigratePluginDataToParentPluginMetadata(pluginsMetadata []*repository.PluginMetadata) error {
+	dbConnection := impl.globalPluginRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		impl.logger.Errorw("MigratePluginDataToParentPluginMetadata, error in beginning transaction", "err", err)
+		return err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	pluginMetadataToUpdate := make([]*repository.PluginMetadata, 0)
+	identifierVsPluginMetadata := make(map[string]*repository.PluginMetadata)
+	for _, item := range pluginsMetadata {
+		identifier := CreateUniqueIdentifier(item.Name, 0)
+		if _, ok := identifierVsPluginMetadata[identifier]; ok {
+			identifier = CreateUniqueIdentifier(item.Name, item.Id)
+		}
+		identifierVsPluginMetadata[identifier] = item
+	}
+
+	for identifier, pluginMetadata := range identifierVsPluginMetadata {
+		pluginParentMetadata, err := impl.globalPluginRepository.GetPluginParentMetadataByIdentifier(identifier)
+		if err != nil {
+			impl.logger.Errorw("MigratePluginDataToParentPluginMetadata, error in GetPluginParentMetadataByIdentifier", "pluginIdentifier", identifier, "err", err)
+			return err
+		}
+		if pluginParentMetadata.Id > 0 {
+			//plugin data already migrated
+			continue
+		}
+		parentMetadata := repository.NewPluginParentMetadata()
+		parentMetadata.SetParentPluginMetadata(pluginMetadata).CreateAuditLog(bean.SystemUserId)
+		parentMetadata.Identifier = identifier
+		parentMetadata, err = impl.globalPluginRepository.SavePluginParentMetadata(tx, parentMetadata)
+		if err != nil {
+			impl.logger.Errorw("MigratePluginDataToParentPluginMetadata, error in saving plugin parent metadata", "err", err)
+			return err
+		}
+		pluginMetadata.PluginParentMetadataId = parentMetadata.Id
+		pluginMetadataToUpdate = append(pluginMetadataToUpdate, pluginMetadata)
+
+	}
+
+	if len(pluginMetadataToUpdate) > 0 {
+		err = impl.globalPluginRepository.UpdatePluginMetadataInBulk(pluginMetadataToUpdate, tx)
+		if err != nil {
+			impl.logger.Errorw("MigratePluginDataToParentPluginMetadata, error in updating plugin metadata in bulk", "err", err)
+			return err
+		}
+	}
+	err = tx.Commit()
+	if err != nil {
+		impl.logger.Errorw("MigratePluginDataToParentPluginMetadata, error in committing db transaction", "err", err)
+		return err
+	}
+	return nil
+}
+
+/*
+GetPluginDetailV2
+if len(pluginIds)>0, this returns all the details of all the plugin versions provided
+if len(parentPluginIds)>0 && isLatest=false, this returns all the details of all the versions in a plugin.
+if len(parentPluginIds)>0 && isLatest=true, this returns all the details of the latest version of a plugin only and minimal data for all the non-latest versions.
+*/
+func (impl *GlobalPluginServiceImpl) GetPluginDetailV2(pluginVersionIds, parentPluginIds []int, fetchLatestVersionDetailsOnly bool) (*PluginsDto, error) {
+	pluginParentMetadataDtos := make([]*PluginParentMetadataDto, 0)
```

Comment:
update len in make and for all other such cases

Reply:
done